scripts/trim_indel_ends.py

```diff
@@ -0,0 +1,452 @@
+"""
+    File name: trim_indel_ends.py
+    Author: Yutong Qiu
+    Date: Nov 20
+    Description: For each pair of alinment between reads, if the best alignment includes indels at the beginning and ends
+        of the reads, then trim off the indels.
+    Input: 
+        - old sets of reads for each sample
+        - aln_stats
+    Output: 
+        - new set of reads for each sample.
+
+    Trimming logic alternatives:
+        - Trim prefix and suffix regardless of the rest of the alignment --- set a threshold
+        - Trim prefix and suffix only when the rest of the alignment is all matches --- set a threshold
+"""
+
+import pandas as pd
+from collections import defaultdict, Counter
+import re
+import edlib
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+from Bio.Seq import Seq
+import matplotlib.pyplot as plt
+from tqdm import tqdm
+import os
+import sys
+import argparse
+import datetime
+
+#--------------------------------------------------------
+# parser
+#--------------------------------------------------------
+# positional/optional arguments
+posList = ['manifest','unique_isoform_aln']
+optList = ['mode', 'threshold']
+sys.stdout.write('\n'); sys.stdout.flush()
+
+# author and version info
+usage = sys.argv[0] + ' <options> ' + ' '.join(posList) + '\n'
+author = 'Author: Yutong Qiu (yutongq@andrew.cmu.edu)'
+version = 'Version: 1.0.0.0 (2022-11-23)'
+description = '\nDescription: This program trims isoform sequences to reduce false positives. Output trimmed sequences in trimmed.fasta'
+
+parser = argparse.ArgumentParser(usage = '\n'.join([usage, author, version, description]), formatter_class=argparse.RawTextHelpFormatter)
+# positional arguments
+parser.add_argument(posList[0], type=str, help='string\tinput manifest csv file in format: sample,fa.  e.g. /in/manifest.csv')
+parser.add_argument(posList[1], type=str, help='string\talignment between similar isoforms within each window.  e.g. /in/unique_isoforms_and_aln_stats.tsv')
+# optional arguments
+parser.add_argument('-m', '--'+optList[0], type=str, metavar='', default='lenient', help='string\tMode of trimming. "Strict" only trims when sequences match exactly after trimming; "Lenient" trims whenever prefix and suffix indel numbers below threshold, defauls to "Lenient"')
+parser.add_argument('-t', '--'+optList[1], type=int, metavar='', default=10000, help='int\tThreshold for trimming, defaults to 10000 (trim every prefix and suffix)')
+
+# strict positional/optional arguments checking
+argsDict = vars(parser.parse_args())
+sys.stdout.write('\n' + datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - ' + sys.argv[0] + ' - Parsing Input Arguements...\n\n'); sys.stdout.flush()
+for key, value in argsDict.items():
+    if key in posList: sys.stdout.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + ' - Required Argument - ' + key +': '+ str(value) + '\n'); sys.stdout.flush()
```

Comment:
This is better written to a logger:

```
logger.info(gument - ' + key +': '+ str(value))
```

The user or developer can configure what additional information is printed to the log, eg time, thread info and whatnot